#goal definition
char[] outputData
---
#result definition
bool isOK
---
#feedback
int32[] sequence